src/Microsoft.AspNetCore.Mvc.Core/Internal/ApiBehaviorApplicationModelProvider.cs

```diff
@@ -250,6 +242,21 @@ internal BindingSource InferBindingSourceForParameter(ParameterModel parameter)
             return bindingSource;
         }
 
+        internal static void DiscoverApiConvention(ActionModel actionModel, ApiConventionTypeAttribute[] apiConventionAttributes)
+        {
+            if (actionModel.Filters.OfType<IApiResponseMetadataProvider>().Any())
+            {
+                // If an action already has providers, don't discover any from conventions.
+                return;
+            }
+
+            var apiConventionItem = ApiConventionResult.GetApiConvention(actionModel.ActionMethod, apiConventionAttributes);
+            if (apiConventionItem != null)
+            {
+                actionModel.Properties[typeof(ApiConventionResult)] = apiConventionItem;
```

Comment:
Is it possible for someone to extend the system and provide their own `ApiConventionResult`s on the action model's properties? Aka, should we verify that `ApiConventionResult` doesn't exist in properties prior to getting them.